### PR TITLE
RIA-3412_Pay_Submit

### PR DESCRIPTION
**Pull request by Lee365**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RIA-3412

### Change description ###
RIA-3412_Pay_Submit

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iacasepaymentsapi/domain/handlers/presubmit/PaymentAppealHandler.java [hunk 1]
```diff
@@ -71,7 +57,7 @@ public boolean canHandle(
         requireNonNull(callback, "callback must not be null");
 
         return (callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT)
-                && (callback.getEvent() == Event.PAYMENT_APPEAL);
+                && (callback.getEvent() == Event.PAY_AND_SUBMIT_APPEAL);
```
Comment (makowalski) [inline src/main/java/uk/gov/hmcts/reform/iacasepaymentsapi/domain/handlers/presubmit/PaymentAppealHandler.java:60]: I would just add `PAY_AND_SUBMIT_APPEAL` event and keep `PAYMENT_APPEAL` as it is. Eventually we need to display PBA account list for both events

File: src/main/java/uk/gov/hmcts/reform/iacasepaymentsapi/domain/handlers/presubmit/PaymentAppealFeePreparer.java [hunk 2]
```diff
@@ -44,7 +44,7 @@ public boolean canHandle(
         return (callbackStage == PreSubmitCallbackStage.ABOUT_TO_START
                && (callback.getEvent() == Event.START_APPEAL
                     || callback.getEvent() == Event.EDIT_APPEAL
-                    || callback.getEvent() == Event.PAYMENT_APPEAL));
+                    || callback.getEvent() == Event.PAY_AND_SUBMIT_APPEAL));
```
Comment (makowalski) [inline src/main/java/uk/gov/hmcts/reform/iacasepaymentsapi/domain/handlers/presubmit/PaymentAppealFeePreparer.java:47]: Let's keep `PAYMENT_APPEAL` too

File: src/main/java/uk/gov/hmcts/reform/iacasepaymentsapi/domain/handlers/presubmit/PayAndSubmitAppealAccountPreparer.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+                .getCaseDetails()
+                .getCaseData();
+
+        List<String> accountsFromOrg = refDataService.getOrganisationResponse()
+            .getOrganisationEntityResponse().getPaymentAccount();
+
+        if (accountsFromOrg.isEmpty()) {
+            PreSubmitCallbackResponse<AsylumCase> response =
+                new PreSubmitCallbackResponse<>(callback.getCaseDetails().getCaseData());
+            response.addErrors(Collections.singleton("There are no payment accounts"));
```
Comment (makowalski) [inline src/main/java/uk/gov/hmcts/reform/iacasepaymentsapi/domain/handlers/presubmit/PayAndSubmitAppealAccountPreparer.java:66]: this error handling could be changed in future depends on Design outcome. It is good that you thought about it

File: src/main/java/uk/gov/hmcts/reform/iacasepaymentsapi/domain/handlers/presubmit/PaymentAppealHandler.java (hunk 1 above)
Comment (Lee365) [reply src/main/java/uk/gov/hmcts/reform/iacasepaymentsapi/domain/handlers/presubmit/PaymentAppealHandler.java]: Done

**Pull request closed without merging**